components/components.go

```diff
@@ -2,144 +2,133 @@
+	return defaultSet
 }
 
-const (
-	StateHealthy      = "Healthy"
-	StateUnhealthy    = "Unhealthy"
-	StateInitializing = "Initializing"
-	StateDegraded     = "Degraded"
-)
+// IsComponentRegistered checks if a component is registered in the default set.
+func IsComponentRegistered(name string) bool {
+	defaultSetMu.RLock()
+	defer defaultSetMu.RUnlock()
 
-type Event struct {
-	Time             metav1.Time              `json:"time"`
-	Name             string                   `json:"name,omitempty"`
-	Type             common.EventType         `json:"type,omitempty"`
-	Message          string                   `json:"message,omitempty"`    // detailed message of the event
-	ExtraInfo        map[string]string        `json:"extra_info,omitempty"` // any extra information the component may want to expose
-	SuggestedActions *common.SuggestedActions `json:"suggested_actions,omitempty"`
+	return isComponentRegistered(defaultSet, name)
 }
 
-type Metric struct {
-	UnixSeconds         int64   `json:"unix_seconds"`
-	MetricName          string  `json:"metric_name"`
-	MetricSecondaryName string  `json:"metric_secondary_name,omitempty"`
-	Value               float64 `json:"value"`
+// isComponentRegistered checks if a component is registered in the default set.
+func isComponentRegistered(set map[string]Component, name string) bool {
+	_, ok := set[name]
+	return ok
 }
 
-type Metrics []Metric
+// GetComponent gets a component from the default set.
+// It returns an error if the component is not found.
+func GetComponent(name string) (Component, error) {
+	defaultSetMu.RLock()
+	defer defaultSetMu.RUnlock()
 
-type Info struct {
-	States  []State `json:"states"`
-	Events  []Event `json:"events"`
-	Metrics Metrics `json:"metrics"`
+	return getComponent(defaultSet, name)
 }
 
-var (
-	defaultSetMu sync.RWMutex
-	defaultSet   = make(map[string]Component)
-)
-
-func IsComponentRegistered(name string) bool {
-	defaultSetMu.RLock()
-	defer defaultSetMu.RUnlock()
+// getComponent gets a component from the default set.
+// It returns an error if the component is not found.
+func getComponent(set map[string]Component, name string) (Component, error) {
+	if set == nil {
+		return nil, fmt.Errorf("component set not initialized: %w", errdefs.ErrUnavailable)
+	}
 
-	_, ok := defaultSet[name]
-	return ok
+	v, ok := set[name]
+	if !ok {
+		return nil, fmt.Errorf("component %s not found: %w", name, errdefs.ErrNotFound)
+	}
+	return v, nil
 }
 
+// RegisterComponent registers a component in the default set.
+// It returns an error if the component is already registered.
 func RegisterComponent(name string, comp Component) error {
 	defaultSetMu.Lock()
 	defer defaultSetMu.Unlock()
 
-	if defaultSet == nil {
+	return registerComponent(defaultSet, comp)
+}
+
+// registerComponent registers a component in the default set.
+// It returns an error if the component is already registered.
+func registerComponent(set map[string]Component, comp Component) error {
```

Comment:
and we don't need it as well...

Reply:
mainly, to make these unit testable, what do you think